pkg/rfc2317/arpa.go

```diff
@@ -0,0 +1,107 @@
+package rfc2317
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+// CidrToInAddr converts a CIDR block into its reverse lookup (in-addr) name.
+// Given "2001::/16" returns "1.0.0.2.ip6.arpa"
+// Given "10.20.30.0/24" returns "30.20.10.in-addr.arpa"
+// Given "10.20.30.0/25" returns "0/25.30.20.10.in-addr.arpa" (RFC2317)
+func CidrToInAddr(cidr string) (string, error) {
+
+	// If cidr is an IP address, turn it into a CIDR by adding /32 or /128
+	ip := net.ParseIP(cidr)
+	if ip != nil {
+		if ip.To4() != nil {
+			cidr = cidr + "/32"
```

Comment:
I think this produces incorrect results for IPv4-mapped IPv6 addresses.

Reply:
Fixed.